src/Streamly/Internal/Data/Array/Foreign.hs

```diff
@@ -195,18 +196,41 @@ fromStream (SerialT m) = P.fold A.write m
 -- Elimination
 -------------------------------------------------------------------------------
 
+{-# INLINE_NORMAL producer #-}
```

Comment:
It looks like we can re-use " producer :: forall m a. (Monad m, Storable a) => Producer m (Array a) a "
from "src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs"

Reply:
In `src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs` we no longer use `unsafeInlineIO`. Till I figure out its implication on the performance, I've just duplicated it here.